cmd/registry/root.go

```diff
@@ -0,0 +1,261 @@
+package registry
+
+import (
+	"fmt"
+	"github.com/astronomer/astro-cli/pkg/input"
+	"github.com/logrusorgru/aurora"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+var (
+	dagVersion      string
+	providerVersion string
+)
+
+func getProviderSearchRoute(providerName string) string {
+	getProviderUrl := url.URL{
+		Scheme:   "https",
+		Host:     registryHost,
+		Path:     fmt.Sprintf("%s/%s", registryApi, providerSearchRoute),
+		RawQuery: fmt.Sprintf(providerSearchQuery, url.QueryEscape(providerName)),
+	}
+	return getProviderUrl.String()
+}
+
+func getProviderRoute(providerId string, providerVersion string) string {
+	getProviderUrl := url.URL{
+		Scheme: "https",
+		Host:   registryHost,
+		Path:   fmt.Sprintf("%s/%s", registryApi, fmt.Sprintf(providerRoute, providerId, providerVersion)),
+	}
+	return getProviderUrl.String()
+}
+
+func addProviderByName(providerName string) {
+	filledProviderRoute := getProviderSearchRoute(providerName)
+
+	providersJson := requestAndGetJsonBody(filledProviderRoute)
+	providers, exists := providersJson["providers"].([]interface{})
+	logKeyNotExists(exists, "providers", providersJson)
+
+	for _, provider := range providers {
+		providerJson := provider.(map[string]interface{})
+
+		providerId, childExists := providerJson["name"].(string) // displayName??
+		logKeyNotExists(childExists, "name", providerJson)
+
+		providerVersion, childExists := providerJson["version"].(string) // displayName??
+		logKeyNotExists(childExists, "name", providerJson)
+
+		addProviderByIdAndVersion(providerId, providerVersion)
+	}
+}
+
+func addProviderByIdAndVersion(providerId string, providerVersion string) {
+	filledProviderRoute := getProviderRoute(providerId, providerVersion)
+	providerJson := requestAndGetJsonBody(filledProviderRoute)
+
+	name, exists := providerJson["name"].(string)
+	logKeyNotExists(exists, "name", providerJson)
+
+	version, exists := providerJson["version"].(string)
+	logKeyNotExists(exists, "version", providerJson)
+
+	addProviderToRequirementsTxt(name, version)
+	// TODO
+	//addProviderExampleDag(name)
+}
+
+// TODO
+//func addProviderExampleDag(name string) {
+//	downloadDag()
+//}
+
+func addProviderToRequirementsTxt(name string, version string) {
+	const filename = "requirements.txt"
+	f, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0655)
+
+	b, err := os.ReadFile(filename)
+	logFatal(err)
+
+	exists := strings.Contains(string(b), name)
+	if exists {
+		log.Infof("%s already exists in %s", name, filename)
+	} else {
+		log.Debugf("Couldn't find %s already in %s", name, string(b))
+		providerString := fmt.Sprintf("%s==%s", name, version)
+		log.Infof("Writing %s to %s", providerString, filename)
+		_, err = f.WriteString(providerString + "\n")
+		logFatal(err)
+		defer f.Close()
+	}
+}
+
+func downloadDag(dagId string, dagVersion string) {
+	// TODO - DAG Version Optional
```

Comment:
for DAG version you're on the right track - you can just default to the keyword "latest" if a version is not provided

Reply:
Nice! That is way easier than pinging the API and getting the first one, which I was about to implement 😅